### DISCUSSION TITLE
Are there any plans to change sort to support multiple fields?

### DISCUSSION THREAD

**Discussion by SimYunSup** [payloadcms/payload#3972] [Feature Requests & Ideas]
Currently, mongoose and drizzle-form support multiple field sort, but defaultSort in payload only allows me to put one field. Do you have any plans to support this, or is there a workaround to allow us to put multiple fields in?

**Comment by DanRibbens** [comment]
Yes, this should be pretty easy to handle in our postgres and mongodb adapters. Not only should defaultSort, we'll also need to make it so that multiple sort is possible with REST, GraphQL and the local API.

I can't think of a workaround to try with how it is written currently.

We can take this one up now that the database implementations are in a good spot.

**Reply by SimYunSup** [reply]
I'm using MongoDB as DB in my project and I've got a temporary workaround via a `pnpm patch`, but I'd like to discuss how to get parameter data type for sortting.

## My temporary workaround

I don't change sort parameter data type(string), but add the code "If sort parameter has space, Consider multiple field sort". Example:

- `color -createdAt` == color desc, createdAt asc

---

But I want to ask if this is a good approach and if it's okay if we don't support a data delivery method like [mongoose](https://mongoosejs.com/docs/api/query.html#Query.prototype.sort()).

And if we decide we don't need a data type like mongoose described above, would it be okay to work on it and raise the PR?

**Reply by r1tsuu** [reply]
Hello, i did PR with this in 1.0, but unfortunately didn't get any reply https://github.com/payloadcms/payload/pull/3236 
Should not be too hard to update with adapter pattern. 
But i have no idea about GraphQL for now,
As workaround now it's also easy to create custom 'find' function that copy-pastes Payload's `find` implementation with overridden `sort` as i done here https://gist.github.com/r1tsuu/4de3e253cde7bf9693bec0a5ca78735a

**Reply by ssyberg** [reply]
bumping this, more robust `sort` without having to override native `find` would be sweet!

**Comment by ssyberg** [comment]
This is a duplicate of https://github.com/payloadcms/payload/discussions/2089, we should consolidate for the upvotes!
